linode/nodepool/framework_resource.go

```diff
@@ -0,0 +1,265 @@
+package nodepool
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/linode/linodego"
+	"github.com/linode/terraform-provider-linode/v2/linode/helper"
+)
+
+func NewResource() resource.Resource {
+	return &Resource{
+		BaseResource: helper.NewBaseResource(
+			helper.BaseResourceConfig{
+				Name:   "linode_lke_node_pool",
+				IDType: types.StringType,
+				Schema: &resourceSchema,
+			},
+		),
+	}
+}
+
+type Resource struct {
+	helper.BaseResource
+}
+
+func (r *Resource) Read(
+	ctx context.Context,
+	req resource.ReadRequest,
+	resp *resource.ReadResponse,
+) {
+	tflog.Debug(ctx, "Read linode_lke_node_pool")
+	var data NodePoolModel
+	client := r.Meta.Client
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
```

Comment:
We'll need to call [`helper.FrameworkAttemptRemoveResourceForEmptyID(...)`](https://github.com/linode/terraform-provider-linode/pull/1314/files#diff-3a5ff5ae9537dc23ec0a3349377aaf1ca0507e8d292f643c99bed33504644d5eR99) here as a workaround for a Crossplane bug we've been encountering 👍

Reply:
done